tests/Main.elm

```diff
@@ -157,6 +157,25 @@ testFromString =
                             ]
                         )
                     )
+        , test "graph with \\l" <|
+            \_ ->
+                Expect.equal
+                    (fromString
+                        (String.join "\n"
+                            [ "digraph hello {"
+                            , "    a [label=\"{A|attr_name : NoneType\\lname : NoneType\\l}\"];"
```

Comment:
Looked at it a little and wanted to point you in the right direction, since it sounds like you might be interested in digging into this. I also haven't looked at this much in a while, so it'll be helpful to map it out and re-familiarize myself.

```
a [label=\"{A|attr_name : NoneType\\lname : NoneType\\l}\"];"
```

This line here is a `node_stmt` which is defined as `node_id [ attr_list ]` in [the grammar]((https://www.graphviz.org/doc/info/lang.html)). Very similarly, it's defined here in code.

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DotLang.elm#L100

`NodeStmt`s get parsed here:

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DotLang.elm#L180-L181

...where the bulk of the work is really handled by `attrList`:

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DotLang.elm#L313-L320

There on line 320, we use `attr` to parse each `Attr` which is really a couple calls to `id` separated by a `=`:

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DotLang.elm#L303-L310

Once we get down to this level, we start getting close to the problem. It doesn't seem like parsing `label` or the `=` is giving us any trouble, but once we get into this double-quoted string with the escaped characters, we run into trouble.

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DotLang.elm#L397-L404

`DQS`, short for double-quoted string, is in another file, and [as this comment suggests](https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DoubleQuoteString.elm#L8), it was taken from an `elm/parser` example. It's not very specific to Dot's use, so it doesn't expect this `\l` in the label.

In the error message you posted in the issue, the error lists a series of things the parser expected to see. That's really this list here:

https://github.com/brandly/elm-dot-lang/blob/fcfc501bfdd084ed7f0f350692c457133a82f389/src/DoubleQuoteString.elm#L22-L30

I think we need to expand this to handle the `\l` too.

Reply:
Thank you for this nice explanation ! I didn't have time to look how to do this, but I really appreciate you takes on your time to explain how you fixed this.